### PR TITLE
Sidecar: Fix process external label on promethues v2.28+ use units.Bytes config type

### LINKED ISSUE #4656: Sidecar: failed to fetch initial external labels. unknown unit
**Thanos, Prometheus and Golang version used**:

**Object Storage Provider**:
COS
**What happened**:
thanos sidecar can not start up.
error message is 
```
2021-09-12T16:09:13.532144772+08:00 level=warn ts=2021-09-12T08:09:13.531921453Z caller=sidecar.go:179 msg="failed to fetch initial external labels. Is Prometheus running? Retrying" err="parse Prometheus config: global:\n  scrape_interval: 30s\n  scrape_timeout: 10s\n  evaluation_interval: 1m\n  external_labels:\n    hash_mod: \"0\"\n    replica: A\n    tenant_id: default\nalerting:\n  alertmanagers:\n  - follow_redirects: true\n    scheme: http\n    timeout: 10s\n    api_version: v2\n    static_configs:\n    - targets: []\nrule_files:\n- /prometheus/recording_rule.yaml\n- /prometheus/alert_rule.yaml\nscrape_configs:\n- job_name: prometheus\n  honor_timestamps: true\n  scrape_interval: 10s\n  scrape_timeout: 10s\n  metrics_path: /metrics\n  scheme: http\n  follow_redirects: true\n  static_configs:\n  - targets:\n    - localhost:9090\n- job_name: etcd_sd\n  honor_timestamps: true\n  scrape_interval: 30s\n  scrape_timeout: 10s\n  metrics_path: /metrics\n  scheme: http\n  body_size_limit: 10485760\n  sample_limit: 15000\n  target_limit: 3000\n  label_name_length_limit: 255\n  label_value_length_limit: 512\n  follow_redirects: true\n: units: unknown unit  in 10485760"
```

prometheus new version instroduce a exempriment config field `body_size_limit`, the type is `units. Base2Bytes` (https://github.com/alecthomas/units). it implement `text.TextUnmarshaler` but not umplement `text.TextMarshaler` (https://github.com/alecthomas/units/blob/master/bytes.go#L43). 
so the marshalled config content can not be marshalled. 

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
Prometheus 2.29.2 + Thanos v0.23.0-rc.0

**Full logs to relevant components**:

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by hanjm**
Fix #4656

Signed-off-by: hanjm <hanjinming@outlook.com>

* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

### CODE REVIEW

File: pkg/promclient/promclient_e2e_test.go [hunk 1]
```diff
@@ -38,12 +40,14 @@ func TestIsWALFileAccessible_e2e(t *testing.T) {
 
 func TestExternalLabels_e2e(t *testing.T) {
 	e2eutil.ForeachPrometheus(t, func(t testing.TB, p *e2eutil.Prometheus) {
-		p.SetConfig(`
-global:
-  external_labels:
-    region: eu-west
-    az: 1
-`)
+		// Keep consist with the config processing in function (*Client).ExternalLabels.
```
Comment (yeya24) [inline pkg/promclient/promclient_e2e_test.go:43]: consistent

**Review by yeya24 — COMMENTED** [review]
Let's also add the latest Prometheus version to our circle CI based unit tests

File: pkg/promclient/promclient_e2e_test.go (hunk 1 above)
Comment (hanjm) [reply pkg/promclient/promclient_e2e_test.go]: Fixed.

**Comment by hanjm** [comment]
> Let's also add the latest Prometheus version to our circle CI based unit tests

Done.

**Review by yeya24 — APPROVED** [review]
LGTM! Thanks for the fix @hanjm

**Pull request merged**